server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -180,8 +182,36 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
 					Lumberjack.info(message, getLumberBaseProperties(documentId, tenantId));
 				}
 			} else {
-				lastCheckpoint = JSON.parse(document.deli);
-			}
+                let checkpoint;
+                let isLocalCheckpoint = false;
+                if(this.localCheckpointEnabled){
+                    // Search local db for checkpoint
+                    Lumberjack.info(`Searching local collection for checkpoint.`, getLumberBaseProperties(documentId, tenantId));
```

Comment:
1. Are you sure you want to log for every single checkpoin?
2. If you want to do so, maybe use a metric instead? And you can set a property like `progress`, which have value of `failedLocalUseGlobal`, `notFondLocalUseGlobal`, `FromLocal` value? which will help us easier to debug in future.

Reply:
For 1, I think having something would be helpful for debugging purposes, so we know where the checkpoints are being written to. I can change this to a metric to be more helpful.

Reply:
Just a note for this, this is not something that would be executed on every checkpoint. It occurs when the deli lambda factory is created, for example when the service restarts and the session has to find the checkpoint from the local (if a session was in progress) or global db. But I will still convert this to a metric, as I think that is a good idea